File: 面试高频题/0003 无重复字符的最长子串.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        mp = {}
        start = -1
        res = 0
        for i in range(len(s)):
            c = s[i]
            if c in mp:
                start = max(start, mp[c])
            res = max(res, i-start)
            mp[c] = i
        return res

File: 面试高频题/0003 无重复字符的最长子串.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> us;
        int n = s.size();
        if (n==0) return 0;
        int res=0;
        int L=0, R=0;
        while (R<n) {
            while (R<n && !us.count(s[R]))
                us.insert(s[R++]);
            res = max(res, R-L);
            us.erase(s[L++]);
        }
        return res;
    }
};